# Lab 1 Initial Analysis

## Setup: import & load data

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.graph_objects as go

df = pd.read_csv('ForeignGifts_edu.csv')


In [ ]:
# df.head()
df['Gift Type'].value_counts()

Gift Type
Contract         17274
Monetary Gift    10936
Real Estate         11
Name: count, dtype: int64

### Initial Analysis 
Looking at the data itself for gift types there's 3. They are contracts, monetary gifts, and real estate. The most amount of gift types are Contracts at 17274 but real estate is the samllest where there is only 11 of them. 

# Q & A 



## Q1 (Which country gives the most money in total?)

In [22]:
df.groupby('Country of Giftor')['Foreign Gift Amount'].sum().sort_values(ascending = False)

Country of Giftor
QATAR           2706240869
ENGLAND         1464906771
CHINA           1237952112
SAUDI ARABIA    1065205930
BERMUDA          899593972
                   ...    
IRAN                    80
ST. LUCIA               50
CROATIA                 26
SLOVAKIA                20
ERITREA                  3
Name: Foreign Gift Amount, Length: 155, dtype: int64

### Answer 
Qatar is the country that gives the most money at 2706240869

## Q2 (Which country initiates the most gifts by count?)

In [17]:
df.groupby('Country of Giftor').size().sort_values(ascending = False)

Country of Giftor
ENGLAND        3655
CHINA          2461
CANADA         2344
JAPAN          1896
SWITZERLAND    1676
               ... 
ERITREA           1
ARMENIA           1
ANTIGUA           1
PALAU             1
ROMANIA           1
Length: 155, dtype: int64

### Answer 
England has the most amount of gifts as 3655

## Q3 (Which country gives the largest gifts on average?)

In [20]:
df.groupby('Country of Giftor')['Foreign Gift Amount'].mean().sort_values(ascending = False)

Country of Giftor
BERMUDA      7.688837e+06
ECUADOR      3.985140e+06
QATAR        3.905109e+06
ANTIGUA      3.435545e+06
JERSEY       3.064589e+06
                 ...     
ST. LUCIA    5.000000e+01
IRAN         4.000000e+01
CROATIA      2.600000e+01
SLOVAKIA     2.000000e+01
ERITREA      3.000000e+00
Name: Foreign Gift Amount, Length: 155, dtype: float64

### Answer 
Bermuda has the highest average at 7.688837e+06

## Q4 (Which institution receive the most money in total?)

In [25]:
df.groupby('Institution Name')['Foreign Gift Amount'].sum().sort_values(ascending = False)

Institution Name
Carnegie Mellon University               1477922504
Cornell University                       1289937761
Harvard University                        954803610
Massachusetts Institute of Technology     859071692
Yale University                           613441311
                                            ...    
Bates College                                250000
Metropolitan State University                235439
University of California, Merced             182514
University of Central Oklahoma               160288
University of Jamestown                         500
Name: Foreign Gift Amount, Length: 318, dtype: int64

### Answer 
Carnegie Mellon University had the most amount of money at 1477922504

## Q5 (Which institution receive the greatest number of gifts by count?)

In [27]:
df.groupby('Institution Name').size().sort_values(ascending = False)

Institution Name
University of California, Los Angeles          3916
California Institute of Technology             2502
Columbia University in the City of New York    1127
Ohio State University (The)                    1014
Johns Hopkins University                        899
                                               ... 
Post University                                   1
Curtis Institute of Music                         1
Washington Adventist University                   1
Corban University                                 1
Des Moines Area Community College                 1
Length: 318, dtype: int64

### Answer
University of California, Los Angeles at 3916

## Q6 (What is the average amount each university receives? Median amount? If they are different why?)

## Q7 (What is the largest flow of a single country to a single institution?)


## Q8 (Which are the top giftors to US academic institutions?)